src/modules/file/file.js

```diff
@@ -94,3 +94,277 @@ function file_entity_field_formatter_view(entity_type, entity, field, instance,
+ */
+function file_field_widget_form(form, form_state, field, instance, langcode, items, delta, element) {
+  try {
+    // Change the item type to a hidden input to hold the file id.
+    // items[delta].type = 'textfield';
+    items[delta].type = 'hidden';
+
+    // If we already have an file for this item, show it.
+    // @TODO: show exisiting fid
+
+    var browse_button_text = t('Add Media');
+    var item_id_base = items[delta].id.replace(/-/g, '_');
+
+    var browse_button_id = items[delta].id + '-add-media-button';
+
+    var html = '<div id="' + items[delta].id + '-media"></div><a href="#" data-role="button" data-icon="camera" ' +
+      'id="' + browse_button_id + '" ' +
+      'data-form_id="' + form.id + '" ' +
+      'data-name="' + element.name + '" ' +
+      'data-delta="' + delta + '" ' +
+      'data-cardinality="' + field.cardinality + '" ' +
+      '>' +
+      browse_button_text +
+      '</a>';
+
+
+
+
+    // Open extra javascript declaration.
+    //
+// $("#" + drupalgap_get_page_id(drupalgap_path_get())).on("pageshow", function () {
+//   document.addEventListener("deviceready", init, false);
+// });
+
+
+    html += '<script type="text/javascript">';
+    // Add device ready listener for PhoneGap camera.
+    html += '$("#' + drupalgap_get_page_id(
+        drupalgap_path_get()) + '").on("pageshow",function(){' +
+      'document.addEventListener(' +
+      '"deviceready", init_media_upload, false );' +
+      '});' ;
+    html += '</script>';
+
+    // Add html to the item's children.
+    if (items[delta].children) {
+      items[delta].children.push({markup: html});
+    } else {
+      items[delta].children = [{markup: html}];
+    }
+
+    //drupalgap_add_js(drupalgap_get_path('module', 'file') + '/file_upload.js');
+  }
+
+  catch (error) {
+    console.log('file_entity_field_formatter_view - ' + error);
+  }
+}
+
+
+/**
+ * Implements hook_assemble_form_state_into_field().
+ * @param {Object} entity_type
+ * @param {String} bundle
+ * @param {String} form_state_value
+ * @param {Object} field
+ * @param {Object} instance
+ * @param {String} langcode
+ * @param {Number} delta
+ * @param {Object} field_key
+ * @return {*}
+ */
+function file_assemble_form_state_into_field(entity_type, bundle,
+                                             form_state_value,
+                                             field,
+                                             instance,
+                                             langcode,
+                                             delta,
+                                             field_key) {
+  try {
+    field_key.value = 'fid';
+    return form_state_value;
+  }
+  catch (error) {
+    console.log('file_assemble_form_state_into_field - ' + error);
+  }
+}
+
+//
+// $("#" + drupalgap_get_page_id(drupalgap_path_get())).on("pageshow", function () {
+//   document.addEventListener("deviceready", init, false);
+// });
+
+function init_media_upload() {
```

Comment:
@luxio This function name should be prefixed with `file_`, that way we can follow some consistency when the API docs are generated.